### PR TITLE
Provided Types virtualization

### PR DESCRIPTION
**Pull request by DedSec256**
Extension Typing Shim + Provided Entities virtualization

### CODE REVIEW

File: tests/service/EditorTests.fs [hunk 1]
```diff
@@ -311,7 +311,7 @@ type Test() =
     let parseResult, typeCheckResults =  parseAndCheckScript(file, input) 
 
     let decls = typeCheckResults.GetDeclarationListInfo(Some parseResult, 4, inputLines.[3], PartialLongName.Empty(20), (fun _ -> []))|> Async.RunSynchronously
-    let item = decls.Items |> Array.tryFind (fun d -> d.Name = "abc")
+    let item = decls.Items |> Seq.tryFind (fun d -> d.Name = "abc")
```
Comment (auduchinok) [inline tests/service/EditorTests.fs]: Is it related to the type providers somehow?

Comment (DedSec256) [reply tests/service/EditorTests.fs]: no and it's fixed in https://github.com/auduchinok/visualfsharp/pull/4

File: src/fsharp/ExtensionTyping.fsi [hunk 2]
```diff
@@ -4,16 +4,16 @@
 
 namespace FSharp.Compiler
 
+open System.Reflection
```
Comment (auduchinok) [inline src/fsharp/ExtensionTyping.fsi]: Why it's not near other opens?

File: src/fsharp/ExtensionTyping.fsi [hunk 3]
```diff
@@ -4,16 +4,16 @@
...
+
 #if !NO_EXTENSIONTYPING
 
-module internal ExtensionTyping =
+module ExtensionTyping =
 
     open System
-    open System.IO
```
Comment (auduchinok) [inline src/fsharp/ExtensionTyping.fsi]: Do we really need this change?

File: src/fsharp/ExtensionTyping.fs [hunk 4]
```diff
@@ -123,58 +123,6 @@ module internal ExtensionTyping =
             // No appropriate constructor found
             raise (TypeProviderError(FSComp.SR.etProviderDoesNotHaveValidConstructor(), typeProviderImplementationType.FullName, m))
 
-    let GetTypeProvidersOfAssembly
```
Comment (auduchinok) [inline src/fsharp/ExtensionTyping.fs:129]: Is it necessary to remove it and not just call it below?

**Review by auduchinok — CHANGES_REQUESTED** [review]
Would it be possible to reduce the diff? We'll have to maintain the changes every time we update FCS, so we need to prevent changes that aren't needed.

**Pull request merged**